### PR TITLE
Fix container delete and add test for it

### PR DESCRIPTION
**Pull request by QiWang19**

### CODE REVIEW

**Comment by QiWang19** [comment]
@martinpitt PTAL :)

File: src/util.js [hunk 1]
```diff
@@ -120,6 +120,10 @@ export function updateContainers() {
         varlinkCall(PODMAN, "io.podman.ListContainers")
                 .then(reply => {
                     let newContainersMeta = reply.containers;
+                    if (newContainersMeta === null) {
+                        resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                        return;
+                    }
```
Comment (martinpitt) [inline src/util.js]: Out of interest, is that related to this PR? It seems odd that ListContainers would ever answer with a null value, instead of an empty list. It's okay to handle it like this, although it would be simpler to provide a fallback value:

    let newContainersMeta = reply.containers || [ ];

File: test/check-application [hunk 2]
```diff
@@ -72,6 +72,23 @@ class TestApplication(testlib.MachineCase):
         # check exited alpine not in running list
         b.wait_not_in_text("#containers-containers", "alpine:latest")
 
+        # delete running container busybox using force delete
+        b.click('#containers-containers tbody tr:contains("busybox:latest") + tr button.btn-delete')
+        self.confirm_modal("btn-danger")
+        self.confirm_modal("btn-danger")
```
Comment (martinpitt) [inline test/check-application]: Having to confirm two consecutive dialogs about the same thing feels strange to me. Of course this behaviour can be fixed in a follow-up PR, i. e. only have a "delete" or "force delete" dialog, not both.

However, in this case the test should make some assertions about the dialog, like verifying the primary button label ("Delete" or "Force Delete"). You could pass the expected button label as an argument to `confirm_modal()`.

File: test/check-application [hunk 3]
```diff
@@ -72,6 +72,23 @@ class TestApplication(testlib.MachineCase):
...
+        b.wait_not_present(".modal-dialog div")
```
Comment (martinpitt) [inline test/check-application]: Right now you have a race condition between the two `confirm_modal()` calls. The function itself doesn't wait for the dialog to close, and you also don't call this in between the two dialogs. So the second call would still see the first dialog before it closes.

I suggest to move this `wait_not_present` into `confirm_modal()`, to avoid forgetting about it.

File: src/util.js [hunk 4]
```diff
@@ -110,7 +110,7 @@ export function format_memory_and_limit(usage, limit) {
 
 // TODO: handle different kinds of errors
 function handleVarlinkCallError(ex) {
-    console.error("Failed to do varlinkcall:", JSON.stringify(ex));
+    console.error("Failed to do varlinkcall:", ex);
```
Comment (KKoukiou) [inline src/util.js]: This is not good change for too reasons:
1rst when you are running the tests, and you are seeing the javascript logs in your terminal, you need JSON.stringify to see the actual object.
2nd this change is not relevant to this PR, it's good approach to keep the PRs do exactly what they mention in the commit message, so that in case something break, it will be easier to find which commit broke it from the history.

Comment (QiWang19) [reply src/util.js]: thanks, I will change it back.

**Comment by QiWang19** [comment]
@martinpitt PTAL

File: test/check-application [hunk 5]
```diff
@@ -72,6 +72,21 @@ class TestApplication(testlib.MachineCase):
...
         b.wait_not_in_text("#containers-containers", "alpine:latest")
 
+        # delete running container busybox using force delete
+        b.click('#containers-containers tbody tr:contains("busybox:latest") + tr button.btn-delete')
+        self.confirm_modal("btn-ctr-delete")
+        self.confirm_modal("btn-ctr-focedelete")
+        b.wait_not_in_text("#containers-containers", "busybox:latest")
+
+        # delete the exited alpine
+        self.filter_continers("all")
```
Comment (martinpitt) [inline test/check-application]: Can you please do a separate PR to fix that typo?
